test/coverage/datamodel/UuidValidatorTest.cpp

```diff
@@ -23,16 +26,216 @@ class TestFixture : public testing::Test {
+	        .count();
+	uint64_t msb =
+	    (8ULL << 12) | (timestamp << 16) | 0x1FFF;  // Version 8, counter = 8191
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	auto [valid, reason] = uprotocol::datamodel::validator::uuid::isUuid(uuid);
+	EXPECT_FALSE(valid);
+}
+
+// Test retrieving version
+TEST_F(TestUuidValidator, RetrieveVersion) {
+	uint64_t msb = (8ULL << 12) | (0x123ULL);  // Version 8, counter = 0x123
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	EXPECT_EQ(uprotocol::datamodel::validator::uuid::getVersion(uuid), 8);
+}
+
+// Test retrieving variant
+TEST_F(TestUuidValidator, RetrieveVariant) {
+	uint64_t msb = (8ULL << 12) | (0x123ULL);  // Version 8, counter = 0x123
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	EXPECT_EQ(uprotocol::datamodel::validator::uuid::getVariant(uuid), 2);
+}
+
+TEST_F(TestUuidValidator, RetrieveTimestamp) {
+	uint64_t timestamp =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count();
+	uint64_t msb =
+	    (8ULL << 12) | (timestamp << 16) | (0x123ULL);  // Version 8, timestamp
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	auto uuid_time = uprotocol::datamodel::validator::uuid::getTime(uuid);
+	auto uuid_timestamp = std::chrono::duration_cast<std::chrono::milliseconds>(
+	                          uuid_time.time_since_epoch())
+	                          .count();
+	EXPECT_EQ(uuid_timestamp, timestamp);
+}
+
+// Test retrieving elapsed time
+TEST_F(TestUuidValidator, RetrieveElapsedTime) {
+	uint64_t timestamp =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count() -
+	    30000;  // 30 seconds in the past
+	uint64_t msb =
+	    (8ULL << 12) | (timestamp << 16) | (0x123ULL);  // Version 8, timestamp
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	EXPECT_GE(
+	    uprotocol::datamodel::validator::uuid::getElapsedTime(uuid).count(),
+	    30000);
+}
+
+// Test retrieving remaining time
+TEST_F(TestUuidValidator, RetrieveRemainingTime) {
+	uint64_t timestamp =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count() -
+	    30000;  // 30 seconds in the past
+	uint64_t msb =
+	    (8ULL << 12) | (timestamp << 16) | (0x123ULL);  // Version 8, timestamp
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	EXPECT_LE(uprotocol::datamodel::validator::uuid::getRemainingTime(
```

Comment:
How much less than 30s is allowed?

Reply:
sounds good, I'll update this with 10 second tolerance